### PR TITLE
add precommit deposit and garbage collection

### PR DESCRIPTION
**Pull request by zixuanzh**
**In this PR**

- [x] Add PRECOMMIT_DEPOSIT
- [x] Miners need to include at least that amount in PreCommitSector
- [x] Miners get back the PRECOMMIT_DEPOSIT at ProveCommitSector
- [x] PRECOMMIT_DEPOSIT will be burned when PreCommit expires before receiving a ProveCommit
- [x] Burn PRECOMMIT_DEPOSIT at `_expirePreCommittedSectors` and clear inactive deals in PreCommittedSectors as a form of garbage collection

### CODE REVIEW

File: src/systems/filecoin_mining/storage_mining/constants.go [hunk 1]
```diff
@@ -11,3 +14,7 @@ const PROVING_PERIOD = block.ChainEpoch(2)                    // placeholder
 const SURPRISE_CHALLENGE_FREQUENCY = 2 // placeholder
 // how far into their PP does a miner get their first challenge
 const SUPRISE_NO_CHALLENGE_PERIOD = PROVING_PERIOD / SURPRISE_CHALLENGE_FREQUENCY
+
+// FIL deposit per sector precommit in Interactive PoRep
+// refunded after ProveCommit but burned if PreCommit expires
+const PRECOMMIT_DEPOSIT = actor.TokenAmount(0) // placeholder
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/constants.go]: Should this be a function of sector size, so a deposit per byte or per GiB?

File: src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go [hunk 2]
```diff
@@ -580,19 +580,21 @@ func (a *StorageMinerActorCode_I) _verifySeal(rt Runtime, onChainInfo sector.OnC
...
-
 	// can be called regardless of Challenged status
 
-	// TODO: might take collateral in case no ProveCommit follows within sometime
-	// TODO: collateral also penalizes repeated precommit to get randomness that one likes
 	// TODO: might be a good place for Treasury
 
 	h, st := a.State(rt)
 
+	msgValue := rt.ValueReceived()
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go]: I think it's worth documenting somewhere up the top that the balance of a storage miner actor should equal exactly the sum of precommit deposits not yet either returned or burnt.

Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go]: ack, will do

File: src/systems/filecoin_mining/storage_mining/constants.go (hunk 1 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/constants.go]: Not sure, but think it's a good suggestion, I can add that for now, simple change to add/remove later.

Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/constants.go]: cc @jzimmerman @nicola

**Pull request merged**